### PR TITLE
PROXY variables documentation

### PR DESCRIPTION
**Pull request by vpodzime**

### CODE REVIEW

**Review by TheMeaningfulEngineer — COMMENTED** [review]
image


Renders well.
I've only ever tested this with `HTTPS_PROXY`.
Pipeline complains over dead links.

File: 03.Client-installation/07.Configuration/docs.md [hunk 1]
```diff
@@ -39,13 +39,19 @@ The mechanism for providing the configuration file and specifying the configurat
...
-| Data directory          | `/usr/share/mender` | `MENDER_DATA_DIR`      |
-| Persistent datastore    | `/var/lib/mender`   | `MENDER_DATASTORE_DIR` |
+| Environment variable   | Description                             | Default value       |
+| ---------------------- | --------------------------------------- | ------------------- |
+| `MENDER_CONF_DIR`      | Configuration directory                 | `/etc/mender`       |
+| `MENDER_DATA_DIR`      | Data directory                          | `/usr/share/mender` |
+| `MENDER_DATASTORE_DIR` | Persistent datastore                    | `/var/lib/mender`   |
+| `HTTP_PROXY`           | Proxy server to use for HTTP            | empty               |
+| `HTTPS_PROXY`          | Proxy server to use for HTTPS           | empty               |
+| `NO_PROXY`             | Hosts that should not go through proxy  | empty               |
```
Comment (nickanderson) [inline 03.Client-installation/07.Configuration/docs.md:53]: Thank you!

**Review by nickanderson — APPROVED** [review]
Very nice improvement, thank you

**Comment by vpodzime** [comment]
> Pipeline complains over dead links.

Yup, that's because we need https://github.com/mendersoftware/mender-docs-site/pull/499#event-14229809307 together with this. One without the other will fail.

**Review by lluiscampos — APPROVED** [review]
Thank you very much for taking this one :pray: 

> > Pipeline complains over dead links.
> 
> Yup, that's because we need [mendersoftware/mender-docs-site#499](https://github.com/mendersoftware/mender-docs-site/pull/499) together with this. One without the other will fail.

No, that is independent - it is a nice to have redirect (thanks for adding it!) but the CI failure is about internal links within the docs content.

We just missed this one:
```
[lluis@gerty mender-docs]((HEAD detached at FETCH_HEAD))$ git diff
diff --git a/06.Artifact-creation/04.State-scripts/docs.md b/06.Artifact-creation/04.State-scripts/docs.md
index 41ebe7ec..8efff130 100644
--- a/06.Artifact-creation/04.State-scripts/docs.md
+++ b/06.Artifact-creation/04.State-scripts/docs.md
@@ -87,7 +87,7 @@ All other return codes are reserved for future use by Mender and should not be u
 
 ### Retry later
 
-State scripts are allowed to return a specific error code (`21`), in which case the client will sleep for a time configured by [StateScriptRetryIntervalSeconds](../../03.Client-installation/07.Configuration/50.Configuration-options/docs.md#statescriptretryintervalseconds) before the state script is called again. Note that scripts are not allowed to retry for infinitely long. Please see description of [StateScriptRetryTimeoutSeconds](../../03.Client-installation/07.Configuration-file/50.Configuration-options/docs.md#statescriptretrytimeoutseconds) for more information.
+State scripts are allowed to return a specific error code (`21`), in which case the client will sleep for a time configured by [StateScriptRetryIntervalSeconds](../../03.Client-installation/07.Configuration/50.Configuration-options/docs.md#statescriptretryintervalseconds) before the state script is called again. Note that scripts are not allowed to retry for infinitely long. Please see description of [StateScriptRetryTimeoutSeconds](../../03.Client-installation/07.Configuration/50.Configuration-options/docs.md#statescriptretrytimeoutseconds) for more information.
 
 This feature is useful e.g when you want user confirmation before proceeding with the update as is described in the [Update confirmation by end user](#update-confirmation-by-end-user) section on this page.

```

**Comment by vpodzime** [comment]
> We just missed this one:

Actually, there was one more:
```

File: 03.Client-installation/02.Install-with-Debian-package/01.Upgrading/docs.md [hunk 2]
```diff
@@ -45,7 +45,7 @@ The following features and config are not available in Mender client 4.0.0 and l
   updates](https://docs.mender.io/3.6/overview/customize-the-update-process#synchronized-updates)
 - [Update Management API](https://docs.mender.io/3.6/device-side-api/io.mender.update1)
 - The
-  [connectivity](https://docs.mender.io/3.6/client-installation/configuration-file/configuration-options#connectivity)
+  [connectivity](https://docs.mender.io/3.6/client-installation/configuration/configuration-options#connectivity)
```
Comment (lluiscampos) [inline 03.Client-installation/02.Install-with-Debian-package/01.Upgrading/docs.md:48]: These are special links for a legacy version (3.6) so you should not update them.

The patch that I posted was enough. I promise - I run the checks locally :wink:

Comment (lluiscampos) [reply 03.Client-installation/02.Install-with-Debian-package/01.Upgrading/docs.md]: I don't know... I didn't find that one on my local checkout, but I might have fetched from your branch at a different time or something.

It works now for CI. And I checked again locally with the latest fetch and looks good here too. So... :+1: :shrug:

**Review by lluiscampos — APPROVED** [review]

**Comment by lluiscampos** [comment]
@mender-test-bot cherry-pick to:
* 3.7.x
* hosted

**Pull request merged**